applications/plugins/SofaImplicitField/python/src/Binding_ScalarField.cpp

```diff
@@ -33,22 +33,67 @@ namespace sofaimplicitfield {
 using namespace sofapython3;
 using sofa::component::geometry::ScalarField;
 using sofa::core::objectmodel::BaseObject;
-using sofa::type::Vec3d;
+using sofa::type::Vec3;
+using sofa::type::Mat3x3;
 
 class ScalarField_Trampoline : public ScalarField {
 public:
     SOFA_CLASS(ScalarField_Trampoline, ScalarField);
 
-    double getValue(Vec3d& pos, int& domain) override{
+    // Override this function so that it returns the actual python class name instead of
+    // "ScalarField_Trampoline" which correspond to this utility class.
+    std::string getClassName() const override
+    {
+        PythonEnvironment::gil acquire;
+
+        // Get the actual class name from python.
+        return py::str(py::cast(this).get_type().attr("__name__"));
+    }
+
+    double getValue(Vec3& pos, int& domain) override
+    {
+        SOFA_UNUSED(domain);
+        PythonEnvironment::gil acquire;
+
+        PYBIND11_OVERLOAD_PURE(double, ScalarField, getValue, pos);
+    }
+
+    Vec3 getGradient(Vec3& pos, int& domain) override
+    {
         SOFA_UNUSED(domain);
         PythonEnvironment::gil acquire;
 
-        PYBIND11_OVERLOAD_PURE(double, ScalarField, getValue, pos.x(), pos.y(), pos.z());
+        PYBIND11_OVERLOAD(Vec3, ScalarField, getGradient, pos);
+    }
+
+    void getHessian(Vec3 &pos, Mat3x3& h) override
+    {
+        /// The implementation for the getHessian is a bit complex as we change de signature between
+        /// the c++ part of the code and python one.
+        // In python, the results is return by the function while in c++ it is part of the function.
```

Comment:
```suggestion
        /// The implementation is a bit more complex compared to getGradient. This is because we change de signature between the c++ API and the python one.
```